leshan-lwm2m-core/src/main/java/org/eclipse/leshan/core/json/JsonRootObject.java

```diff
@@ -35,7 +35,7 @@ public class JsonRootObject {
 
     public JsonRootObject(String baseName, List<JsonArrayEntry> jsonArray, BigDecimal baseTime) {
         this.baseName = baseName;
-        this.jsonArray = jsonArray;
```

Comment:
If you really want to make it immutable. The collection pass to `Collections.unmodifiableList` should not be accessible or people who create the `JsonRootObject` could still change the content of the collection.

So generally we do something like : `Collections.unmodifiableList(new ArrayList(jsonArray))`

Reply:
OK clear - Thanks!